File: src/main/java/fr/lewon/bot/manager/mappers/BotPropertyMapper.java
package fr.lewon.bot.manager.mappers;

import java.util.List;

import org.mapstruct.Mapper;

import fr.lewon.bot.manager.entities.BotPropertyDescriptorDTO;
import fr.lewon.bot.props.BotPropertyDescriptor;

@Mapper(componentModel = "spring")
public interface BotPropertyMapper {

	List<BotPropertyDescriptorDTO> botPropertiesToDto(List<BotPropertyDescriptor> botMethod);

	BotPropertyDescriptorDTO botPropertyToDto(BotPropertyDescriptor botMethod);

}

File: src/main/kotlin/fr/lewon/bot/manager/mappers/BotPropertyMapper.kt
package fr.lewon.bot.manager.mappers

import fr.lewon.bot.manager.entities.BotPropertyDescriptorDTO
import fr.lewon.bot.runner.bot.props.BotPropertyDescriptor
import org.mapstruct.Mapper

@Mapper(componentModel = "spring")
interface BotPropertyMapper {

    fun botPropertiesToDto(botMethod: List<BotPropertyDescriptor>): List<BotPropertyDescriptorDTO>
    fun botPropertyToDto(botMethod: BotPropertyDescriptor): BotPropertyDescriptorDTO

}